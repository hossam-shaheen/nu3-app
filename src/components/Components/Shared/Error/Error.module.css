.error-message,
.no-results {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 15px;
    margin-bottom: 15px;
    font-size: 28px;
    width: 100%;
}

.error-message {
    color: var(--error);
}

.error-message i {
    color: var(--error);
    margin-right: 8px;
}


.no-results {
    color: var(--darkGray);
}