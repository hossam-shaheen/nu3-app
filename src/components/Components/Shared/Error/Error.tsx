import { FunctionComponent } from "react";
import classes from "./Error.module.css";

export const Error: FunctionComponent<{
  errorClass: string;
  error: { icon?: string; message: string };
}> = ({ errorClass, error }): JSX.Element => {
  return (
    <div className={classes[errorClass]}>
      {error.icon && <i className={error.icon}></i>}
      {error?.message}
    </div>
  );
};
export default Error;
