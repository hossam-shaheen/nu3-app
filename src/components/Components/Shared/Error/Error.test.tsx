import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Error from './Error';

const mockedError = {
    icon: "fas fa-exclamation-triangle",
    message: "Failed to fetch data"
}

const mockedNoResults = {
    message: "No results found"
}

test('Error "Failed to Fetch Data" : Renders', () => {
    render(<Error errorClass="error-message" error={mockedError} />);
    const FailedDataElement = screen.getByText('Failed to fetch data', { exact: false });
    expect(FailedDataElement).toBeInTheDocument();
});


test('Error "No results found" : Renders', () => {
    render(<Error errorClass="error-message" error={mockedNoResults} />);
    const noResultElement = screen.getByText('No results found', { exact: false });
    expect(noResultElement).toBeInTheDocument();
});

test('Error "Failed to Fetch Data": Snapshot', () => {
    const component = renderer.create(<Error errorClass="error-message" error={mockedError} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Error "No results found": Snapshot', () => {
    const component = renderer.create(<Error errorClass="error-message" error={mockedError} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});