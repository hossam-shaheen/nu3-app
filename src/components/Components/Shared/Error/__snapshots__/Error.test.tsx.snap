// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error "Failed to Fetch Data": Snapshot 1`] = `
<div
  className="error-message"
>
  <i
    className="fas fa-exclamation-triangle"
  />
  Failed to fetch data
</div>
`;

exports[`Error "No results found": Snapshot 1`] = `
<div
  className="error-message"
>
  <i
    className="fas fa-exclamation-triangle"
  />
  Failed to fetch data
</div>
`;
